src/plugins/intel_cpu/src/transformations/snippets/aarch64/op/gemm_copy_b.hpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/core/attribute_visitor.hpp"
+#include "snippets/op/memory_access.hpp"
+#include "snippets/shape_inference/shape_inference.hpp"
+#include "snippets/shape_types.hpp"
+
+namespace ov::intel_cpu::aarch64 {
+
+/**
+* @interface GemmCopyB
+* @brief The operation for data repacking of GemmCPU.
+         The CPU Generator uses kleidiAI primitives for generation code of Gemm.
+         kleidiAI requiers data repacking for second input of Gemm.
+* @ingroup snippets
+*/
+class GemmCopyB : public snippets::modifier::MemoryAccess, public ov::op::Op {
+public:
+    OPENVINO_OP("GemmCopyB", "SnippetsOpset");
+
+    GemmCopyB(const Output<Node>& x,
+              const element::Type src_type,
+              const size_t offset_in = 0lu,
+              const size_t offset_out0 = 0lu,
+              const std::vector<size_t>& layout_input = {});
+    GemmCopyB(const Output<Node>& x,
+              const element::Type src_type,
+              const PortDescriptor& desc_in0,
+              const PortDescriptor& desc_out0,
+              const std::vector<size_t>& layout_input = {});
+    GemmCopyB() = default;
+
+    size_t get_offset_in() const {
+        return get_input_offset(0);
+    }
+    size_t get_offset_out() const {
+        return get_output_offset(0);
+    }
+
+    element::Type get_src_element_type() const {
+        return m_src_type;
+    }
+
+    bool visit_attributes(AttributeVisitor& visitor) override;
+    void validate_and_infer_types() override;
+    bool has_evaluate() const override {
+        return false;
+    }
+    std::shared_ptr<Node> clone_with_new_inputs(const OutputVector& new_args) const override;
+
+    virtual bool is_adjusted_output_size() const override {
+        return true;
+    }
+    virtual size_t get_adjusted_output_size(const snippets::VectorDims& shape, const size_t& rank) const override;
+
+    class ShapeInfer : public snippets::IShapeInferSnippets {
+        std::vector<size_t> m_layout{};
+        size_t m_num_outs = 1;
```

Comment:
I see that `GemmCopyB` always has one output. Probably it makes sense to remove this attribute